### PR TITLE
fix(Dialog): prevent close confirmation dialog on outside click

### PR DESCRIPTION
**Pull request by dinarosv**
Disable overlay click for variant `confirmation` on Dialog.

The dialog should still be able to close using the buttons on the inside, and optionally using the close button, so I added a new prop `preventOverlayClose`/`prevent_overlay_close` for Modal (instead of using the `preventClose`).

### CODE REVIEW

**Review by tujoworker — COMMENTED** [review]
Very good work 🙌

File: packages/dnb-eufemia/src/components/dialog/__tests__/Dialog.test.tsx [hunk 1]
```diff
@@ -318,6 +318,15 @@ describe('Dialog', () => {
...
   })
+
+  it('does not close with click on overlay for variant confirmation', () => {
+    const Comp = mount(
+      <Dialog {...props} variant="confirmation" openState="opened" />
+    )
+
+    Comp.find('.dnb-modal__content').simulate('click')
+    expect(Comp.exists('.dnb-dialog__inner')).toBe(true)
+  })
```
[the remark this answers is not in the document]
Comment (dinarosv) [reply packages/dnb-eufemia/src/components/dialog/__tests__/Dialog.test.tsx:329]: Oh yes! I totally forgot about the esc key possibility... Will fix it now 👍

File: packages/dnb-design-system-portal/src/docs/uilib/components/modal/prop-table.md [hunk 2]
```diff
@@ -16,6 +16,7 @@
 | `noAnimationOnMobile` / `no_animation_on_mobile` | _(optional)_ same as `noAnimation`, but gets triggered only if the viewport width is less than `40em`. Defaults to false.                                                                                                                       |
 | `animationDuration` / `animation_duration`       | _(optional)_ Duration of animation open/close in ms. Defaults to 300ms.                                                                                                                                                                         |
 | `preventClose` / `prevent_close`                 | _(optional)_ if set to `true` (boolean or string), then the user can't close the modal.                                                                                                                                                         |
+| `preventOverlayClose` / `prevent_overlay_close`  | _(optional)_ Disable clicking the background overlay to close the modal.                                                                                                                                                                        |
```
[the remark this answers is not in the document]
Comment (dinarosv) [reply packages/dnb-design-system-portal/src/docs/uilib/components/modal/prop-table.md]: Done ✅

File: packages/dnb-eufemia/src/components/dialog/__tests__/Dialog.test.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dinarosv) [reply packages/dnb-eufemia/src/components/dialog/__tests__/Dialog.test.tsx:333]: Done ✅

**Review by tujoworker — APPROVED** [review]
👏

**Pull request merged**